unity-renderer/Assets/DCLPlugins/ECS7/Systems/SceneBoundsCheckerSystem/ECSSceneBoundsCheckerSystem.cs

```diff
@@ -0,0 +1,358 @@
+using DCL.Controllers;
+using DCL.ECS7.InternalComponents;
+using DCL.ECSRuntime;
+using DCL.Models;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace ECSSystems.ECSSceneBoundsCheckerSystem
+{
+    public class ECSSceneBoundsCheckerSystem
+    {
+        private readonly IInternalECSComponent<InternalSceneBoundsCheck> sceneBoundsCheckComponent;
+        private readonly IInternalECSComponent<InternalVisibility> visibilityComponent;
+        private readonly IInternalECSComponent<InternalRenderers> renderersComponent;
+        private readonly IInternalECSComponent<InternalColliders> pointerCollidersComponent;
+        private readonly IInternalECSComponent<InternalColliders> physicsCollidersComponent;
+        private readonly IInternalECSComponent<InternalAudioSource> audioSourceComponent;
+        private readonly IECSOutOfSceneBoundsFeedbackStyle outOfBoundsVisualFeedback;
+
+        public ECSSceneBoundsCheckerSystem(IInternalECSComponent<InternalSceneBoundsCheck> sbcComponent,
+            IInternalECSComponent<InternalVisibility> visibilityComponent,
+            IInternalECSComponent<InternalRenderers> renderersComponent,
+            IInternalECSComponent<InternalColliders> pointerColliderComponent,
+            IInternalECSComponent<InternalColliders> physicsColliderComponent,
+            IInternalECSComponent<InternalAudioSource> audioSourceComponent,
+            bool previewMode = false)
+        {
+            this.sceneBoundsCheckComponent = sbcComponent;
+            this.visibilityComponent = visibilityComponent;
+            this.renderersComponent = renderersComponent;
+            this.pointerCollidersComponent = pointerColliderComponent;
+            this.physicsCollidersComponent = physicsColliderComponent;
+            this.audioSourceComponent = audioSourceComponent;
+
+            outOfBoundsVisualFeedback = previewMode ? new ECSOutOfSceneBoundsFeedback_RedWireframe() : new ECSOutOfSceneBoundsFeedback_EnabledToggle();
```

Comment:
Inject `IECSOutOfSceneBoundsFeedbackStyle` instead of this bool.

Reply:
Agreed, I think that would be a "must change" for tests, so I will leave @pravusjif to change that when he start coding the tests